c10/core/DispatchKey.h

```diff
@@ -281,6 +281,10 @@ enum class DispatchKey : uint8_t {
   Functionalize,
   FuncTorchDynamicLayerFrontMode, // See Note [Out-of-tree vmap+grad prototype]
 
+  // The dispatch key for the meta-init backend which forces all tensors to use
+  // the meta device regardless of their real device.
+  MetaInit,
```

Comment:
I am confused. Wasn't this already proposed in another PR a while back and got discontinued in favor of an approach that uses Python Mode?

Reply:
I think you mean #63601. Well that PR did not use a proper dispatcher backend, but rather relied on a peculiar way that injected the `Meta` dispatch key, while removing the rest of the real backends from the `DispatchKeySet`. It was mostly a draft/experiment to see whether implicit meta initialization would work. As the (preferred) alternative Edward suggested using the Python mode, which I did. Unfortunately although looked promising first, because it is below Autograd and does not have access to some features available only in the C++ API, that approach also did not work out.

You can check out the 3rd PR in this stack that I just published to see how I leverage the meta-init backend for the "meta initialization" part of this work.